### PR TITLE
Event packet

### PR DESCRIPTION
**Pull request by akoen**
Currently test does not complete.

### CODE REVIEW

File: main_window/packet_parser.py [hunk 1]
```diff
@@ -50,6 +52,18 @@ class SubpacketIds(Enum):
 }
 DEVICE_TYPE_TO_ID = {y: x for (x, y) in ID_TO_DEVICE_TYPE.items()}
 
+# Matches enum EventId in FLARE: radio.h
```
Comment (DSep) [inline main_window/packet_parser.py:55]: If possible, might want to use the enum "DataEntryValues" in the data_entry_ids.py

File: main_window/packet_parser.py [hunk 2]
```diff
@@ -34,9 +52,23 @@
...
+    0x00: "LAUNCH",
+    0x01: "STAGE SEPARATION",
+    0x02: "MACH LOCK ENTER",
+    0x03: "MACH LOCK EXIT",
+    0x04: "APOGEE",
+    0x05: "DROGUE DEPLOY",
+    0x06: "MAIN DEPLOY",
+    0x07: "LANDED"
+}
+
```
Comment (DSep) [inline main_window/packet_parser.py:66]: If possible, might want to use the enum "DataEntryValues" in the data_entry_ids.py

File: connections/debug/debug_connection.py [hunk 3]
```diff
@@ -54,6 +54,7 @@ def _run(self) -> None:
                 # full_arr.extend(self.config_mock_set_values())
                 # full_arr.extend(self.message_mock_set_values())
                 full_arr.extend(self.bulk_sensor_mock_random())
+                full_arr.extend(self.event_mock_set_values())
```
Comment (SilentSys) [inline connections/debug/debug_connection.py:57]:
```suggestion
                # full_arr.extend(self.event_mock_set_values())
```
Let's keep this commented out for now to reduce spam in the UI.

File: main_window/data_entry_id.py [hunk 4]
```diff
@@ -45,18 +45,32 @@ class DataEntryIds(Enum):
...
-    PRESSURE_DELAY = auto()
-    INITIAL_DESCENT = auto()
-    FINAL_DESCENT = auto()
-    LANDED = auto()
-    WINTER_CONTINGENCY = auto()
-
-    # TODO Add Events, other info stored
+    STATE_NOMINAL = auto()
+    STATE_NONCRITICAL_FAILURE = auto()
+    STATE_CRITICAL_FAILURE = auto()
```
Comment (SilentSys) [inline main_window/data_entry_id.py:50]: Technically these are for "status" not "state"
```suggestion
    STATUS_NOMINAL = auto()
    STATUS_NONCRITICAL_FAILURE = auto()
    STATUS_CRITICAL_FAILURE = auto()
```

File: main_window/packet_parser.py [hunk 5]
```diff
@@ -179,18 +197,14 @@ def message(self, byte_stream: BytesIO, header: Header):
...
-        :return:
-        :rtype:
-        """
         data: Dict = {}
-        # TODO Enumerate event id -> strings, and convert to human readable string here?
-        data[DataEntryIds.EVENT] = byte_stream.read(1)[0]
+        event_bytes = byte_stream.read(2);
+        event_int = self.bytestoint(event_bytes)
+        data_entry_value = EVENT_IDS[event_int]
+        data[DataEntryIds.EVENT] = data_entry_value.value
```
Comment (SilentSys) [inline main_window/packet_parser.py:204]: The pythonic way of using enums (especially those with auto values) is without the `.value`. We also need that to discern between enum values and float/numeric values when we handle the data.
```suggestion
        data[DataEntryIds.EVENT] = data_entry_value
```

File: tests/integration_tests/test_debug.py [hunk 6]
```diff
@@ -133,6 +134,19 @@ def test_single_sensor_packet(qtbot, single_connection_tantalus):
...
+    connection = app.connections['DEBUG_CONNECTION']
+    packet = radio_packets.event(0xFFFFFFFF, 0x00)
+    snapshot = get_event_stats_snapshot()
+    connection.receive(packet)
+
+    assert EVENT_EVENT.wait(snapshot) == 1
+    assert app.rocket_data.last_value_by_device(DeviceType.TANTALUS_STAGE_1_FLARE,
+                                                DataEntryIds.TIME) == 0xFFFFFFFF
+    assert app.rocket_data.last_value_by_device(DeviceType.TANTALUS_STAGE_1_FLARE,
+                                                DataEntryIds.EVENT) == DataEntryValues.EVENT_ARMED.value
```
Comment (SilentSys) [inline tests/integration_tests/test_debug.py:148]: As per the other comment.
```suggestion
                                                DataEntryIds.EVENT) == DataEntryValues.EVENT_ARMED
```

File: tests/integration_tests/test_sim.py [hunk 7]
```diff
@@ -52,11 +52,13 @@ def test_arming(self, qtbot, sim_app, device_type):
         flush_packets(sim_app, device_type)
 
         assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.STATE) == 1
+        assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.EVENT) == DataEntryValues.EVENT_ARMED.value
 
         sim_app.send_command(device_type.name + ".disarm")
         flush_packets(sim_app, device_type)
 
         assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.STATE) == 0
+        assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.EVENT) == DataEntryValues.EVENT_DISARMED.value
```
Comment (SilentSys) [inline tests/integration_tests/test_sim.py:61]: As per the other comment.
```suggestion
        assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.EVENT) == DataEntryValues.EVENT_ARMED

        sim_app.send_command(device_type.name + ".disarm")
        flush_packets(sim_app, device_type)

        assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.STATE) == 0
        assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.EVENT) == DataEntryValues.EVENT_DISARMED
```

**Review by SilentSys — CHANGES_REQUESTED** [review]
Looks great! Really digging the DataEntryValues renaming to make them more specific.

Just a few minor things.

File: main_window/data_entry_id.py (hunk 4 above)
Comment (SilentSys) [reply main_window/data_entry_id.py]: CI is failing. This still needs to be renamed somewhere else in the code.

**Review by SilentSys — APPROVED** [review]
LGTM!

**Review by DSep — APPROVED** [review]
LGTM 👍

File: tests/integration_tests/test_sim.py [hunk 8]
```diff
@@ -52,11 +52,13 @@ def test_arming(self, qtbot, sim_app, device_type):
         flush_packets(sim_app, device_type)
 
         assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.STATE) == 1
+        assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.EVENT) == DataEntryValues.EVENT_ARMED
```
Comment (DSep) [inline tests/integration_tests/test_sim.py:55]: I jumped the gun it seems; the SIM arming test (test_arming) is actually failing on this assertion. All other tests pass.

**Review by DSep — CHANGES_REQUESTED** [review]
It seems like the GS never actually receives an event subpacket, as a result of the arm command.  Not sure if FW or send/receive issue on GS.

**Review by DSep — APPROVED** [review]

**Comment by SilentSys** [comment]
CI failed due to a possible known instability. I initiated a re-run.

**Comment by SilentSys** [comment]
[Implement Event handling in Packet Parsers and general GS usage](https://trello.com/c/IEFftNFs/277-implement-event-handling-in-packet-parsers-and-general-gs-usage)

**Pull request merged**